soc/mediatek/mtk_adsp/mtk_adsp_load.py

```diff
@@ -102,40 +193,35 @@ def main():
         boot_vector = le4(dat[8:12])
         sram = dat[12:12+sram_len]
         dram = dat[12 + sram_len:]
-        assert len(sram) <= MAPPINGS["sram"][1]
-        assert len(dram) <= MAPPINGS["dram"][1]
+        assert len(sram) <= mmio["sram"][1]
+        assert len(dram) <= mmio["dram1"][1]
+
+        # Stop the device and write the regions.  Note that we don't
+        # zero-fill SRAM, as that's been observed to reboot the host
+        # (!!) on mt8186 when the writes near the end of the 512k
+        # region.
+        # pylint: disable=consider-using-enumerate
         for i in range(sram_len):
             maps["sram"][i] = sram[i]
-        for i in range(sram_len, MAPPINGS["sram"][1]):
-            maps["sram"][i] = 0
+        #for i in range(sram_len, mmio["sram"][1]):
+        #    maps["sram"][i] = 0
```

Comment:
is this not needed?

Reply:
Zephyr will zero its own bss after boot, so no.  It's a hygiene thing only (to guarantee that all memory looks consistent, even things like .noinit sections), but disabled because on one of the boards I have it appears to glitch the bus and reboot the whole device.